operator/controller/src/test/java/io/apicurio/registry/operator/unit/CorsTest.java

```diff
@@ -0,0 +1,98 @@
+
+import java.nio.charset.Charset;
+import java.util.LinkedHashMap;
+import java.util.Set;
+
+public class CorsTest {
+
+    private static final String DEFAULT = """
+            apiVersion: registry.apicur.io/v1
+            kind: ApicurioRegistry3
+            metadata:
+              name: simple
+            spec: {}
+            """;
+
+    private static final String WITH_INGRESS = """
+            apiVersion: registry.apicur.io/v1
+            kind: ApicurioRegistry3
+            metadata:
+              name: simple
+            spec:
+              app:
+                ingress:
+                  host: simple-app.apps.cluster.example
+              ui:
+                ingress:
+                  host: simple-ui.apps.cluster.example
+            """;
+
+    private static final String WITH_ENV_VAR = """
+            apiVersion: registry.apicur.io/v1
+            kind: ApicurioRegistry3
+            metadata:
+              name: simple
+            spec:
+              app:
+                env:
+                  - name: QUARKUS_HTTP_CORS_ORIGINS
+                    value: https://ui.example.org
+            """;
+
+    private static final String WITH_ENV_VAR_AND_INGRESS = """
+            apiVersion: registry.apicur.io/v1
+            kind: ApicurioRegistry3
+            metadata:
+              name: simple
+            spec:
+              app:
+                ingress:
+                  host: simple-app.apps.cluster.example
+                env:
+                  - name: QUARKUS_HTTP_CORS_ORIGINS
+                    value: https://ui.example.org
+              ui:
+                ingress:
+                  host: simple-ui.apps.cluster.example
+            """;
+
+    @Test
+    public void testConfigureAllowedOrigins() throws Exception {
+        doTestAllowedOrigins(DEFAULT, "*");
+        doTestAllowedOrigins(WITH_INGRESS, "http://simple-ui.apps.cluster.example",
+                "https://simple-ui.apps.cluster.example");
+        doTestAllowedOrigins(WITH_ENV_VAR, "https://ui.example.org");
+        doTestAllowedOrigins(WITH_ENV_VAR_AND_INGRESS, "http://simple-ui.apps.cluster.example",
+                "https://simple-ui.apps.cluster.example", "https://ui.example.org");
+    }
+
+    private void doTestAllowedOrigins(String cr, String... values) {
+        ApicurioRegistry3 registry = Mapper.deserialize(cr, ApicurioRegistry3.class);
+
+        LinkedHashMap<String, EnvVar> envVars = new LinkedHashMap<>();
+        Cors.configureAllowedOrigins(registry, envVars);
+        Assertions.assertThat(envVars.keySet()).contains(EnvironmentVariables.QUARKUS_HTTP_CORS_ORIGINS);
+        String allowedOriginsValue = envVars.get(EnvironmentVariables.QUARKUS_HTTP_CORS_ORIGINS).getValue();
+        Set<String> allowedOrigins = Set.of(allowedOriginsValue.split(","));
+        Assertions.assertThat(allowedOrigins).containsExactlyInAnyOrder(values);
+    }
+
+    public static String load(String path) {
```

Comment:
Similar functionality is already in `io.apicurio.registry.operator.resource.ResourceFactory#load`. Can we a single place for this functionality? I don't have a preference for where to put it.

Reply:
The resource loading in `ResourceFactory` didn't work from a unit test, I assume due to the difference in thread context classloader.  I can consolidate as long as you're OK with multiple variants of `load()` for the different use cases.  For the unit test I will need to pass a classloader to `load()` so it can properly load the resource.